src/v2/BaseLoginRouter.js

```diff
@@ -160,18 +210,20 @@ export default Router.extend({
     // introspect stateToken when widget is bootstrap with state token
     // and remove it from `settings` afterwards as IDX response always has
     // state token (which will be set into AppState)
-    if (this.settings.get('stateToken') || this.settings.get('proxyIdxResponse')) {
+    if (this.settings.get('oieEnabled')) {
       const idxRespPromise = this.settings.get('proxyIdxResponse') ?
-        handleProxyIdxResponse(this.settings) : introspectStateToken(this.settings);
+        handleProxyIdxResponse(this.settings) : startLoginFlow(this.settings);
       return idxRespPromise
         .then(idxResp => {
           this.settings.unset('stateToken');
           this.settings.unset('proxyIdxResponse');
+          this.settings.unset('useInteractionCodeFlow');
```

Comment:
Can you please create a new `unsetAll` method as well, that can take an array and unset all..